htdocs/compta/paiement/class/paiement.class.php

```diff
@@ -276,17 +276,23 @@ public function create($user, $closepaidinvoices = 0, $thirdparty = null)
 				return -1;
 			}
 			if (empty($currencyofpayment)) {
-				$currencyofpayment = $this->multicurrency_code[$key];
+				if (array_key_exists($key, $this->multicurrency_code)) {
+					$currencyofpayment = $this->multicurrency_code[$key];
+				} else {
+					// Handle the case where the key does not exist in the array
```

Comment:
If $this->multicurrency_code[$key] is not defined, then currencyofpayment is set to null with old code and continue.
With new one, the code is finished.

Code must be similar after and before.